src/solvers/nonlinear_solid_solver.hpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2019, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef NONLINSOLID_SOLVER
+#define NONLINSOLID_SOLVER
+
+#include "mfem.hpp"
+#include "base_solver.hpp"
+
+// Forward declaration
+class NonlinearSolidQuasiStaticOperator;
+class NonlinearSolidDynamicOperator;
```

Comment:
Yes - This should include both the `DynamicSolver` and `QuasiStaticSolver` capabilities when it's done (similar to `ThermalSolver`). This is still very much a work in progress, however.

Reply:
Any objection to just making it "SolidMechanicsSolver" or similar? The other Solver objects don't distinguish between linear/nonlinear.

Reply:
That's a good point. I'm not sure whether we should combine nonlinear operators with linear ones as the underlying data structures are fairly different. However from a user perspective, I can see it being clearer with just a single interface and a flag to set whether it's linear or not. This is essentially what I'm doing with the static and dynamic options anyway.

Reply:
I think I should keep it as it is for now since this PR is getting big enough as is. I'll add another issue to unify the nonlinear and linear implementations.